src/index.js

```diff
@@ -3,13 +3,36 @@ import ReactDOM from "react-dom";
 import "./index.css";
 import App from "./App";
 import { makeServer } from "./server";
+import { BrowserRouter as Router } from "react-router-dom";
+import {
+  AuthProvider,
+  CategoryProvider,
+  FiltersProvider,
+  PlaylistProvider,
+  VideosProvider,
+  WatchLaterProvider,
+} from "./frontend/context/providers";
 
 // Call make Server
 makeServer();
 
 ReactDOM.render(
   <React.StrictMode>
-    <App />
+    <Router>
+      <CategoryProvider>
+        <AuthProvider>
+          <VideosProvider>
+            <FiltersProvider>
+              <PlaylistProvider>
+                <WatchLaterProvider>
+                  <App />
+                </WatchLaterProvider>
+              </PlaylistProvider>
+            </FiltersProvider>
+          </VideosProvider>
+        </AuthProvider>
+      </CategoryProvider>
+    </Router>

```

Comment:
you can create a single provider and put all of these providers in that file, and here you can wrap <App/> component with that single provider to make App.js cleaner.